### PR TITLE
Created Subscribe function

### PR DESCRIPTION
**Pull request by Joshua-Wright76**
Subscribe function takes a callback function and a name and subscribes the callback function to the siloNode at that name. It requires the silo from combineNodes.js, then recursively searches the silo to find the node with the name passed in, and pushes the function into its "modifiers" array.

### CODE REVIEW

**Review by hlmitchell — APPROVED** [review]

File: subscribe.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+                let foundNode = searchSilo(children[i], name);
+                console.log('coming back up. foundNode is: ', foundNode);
+                if(!!foundNode){return foundNode};
+            }
+        }
+    }
+
+    let foundNode = searchSilo(silo, name);
+    foundNode._subscribers.push(component)
+    return foundNode;
```
Comment (G0ldenSp00n-zz) [inline subscribe.js]: The foundNode shouldn't be returned, the subscribe function should return a function that will when called remove the subscribed function from the list, unsubscribing it.

File: subscribe.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
+const silo = require('./combineNodes.js').silo;
+
+const subscribe = (component, name) => {
```
Comment (G0ldenSp00n-zz) [inline subscribe.js]: The subscribe function should be a method on silo, as to not require the importing of the function, and the creation of multiple Silos.

**Review by G0ldenSp00n-zz — CHANGES_REQUESTED** [review]

File: subscribe.js (hunk 1 above)
Comment (Joshua-Wright76) [reply subscribe.js]: You're right. this was an artifact of my internal testing.

**Review by G0ldenSp00n-zz — APPROVED** [review]

File: combineNodes.js [hunk 3]
```diff
@@ -192,6 +192,40 @@ combineNodes = (...args) => {
 
 // ==========> END TESTS that calling a parent function will modify its child for nested objects <========== \\
 
+silo.prototype.subscribe = (component, name) => {
```
Comment (G0ldenSp00n) [inline combineNodes.js]: Silo won't be the silo of the file that imports this because of the module that rescopes this file on it's own.

**Review by G0ldenSp00n-zz — CHANGES_REQUESTED** [review]
Silo won't be the silo of the importing file

**Review by G0ldenSp00n-zz — APPROVED** [review]

**Review by ntsmitty — APPROVED** [review]

**Pull request merged**